vike-vue/renderer/onRenderHtml.ts

```diff
@@ -25,6 +23,17 @@ async function onRenderHtml(pageContext: PageContextServer) {
   const { favicon } = pageContext.config
   const faviconTag = !favicon ? '' : escapeInject`<link rel="icon" href="${favicon}" />`
 
+  let headHtml = ''
+  if (pageContext.config.Head !== undefined) {
+    const headContext: PageContextServer = {
+      ...pageContext,
```

Comment:
Is this OK to shallow-copy pageContext or can this go wrong? I'm making sure not to modify the original pageContext below because of the shallow-copy, but maybe this is still not good enough?

Reply:
Technically there is no issue but, for the sake of separation of concerns, it's usually discouraged, see https://vite-plugin-ssr.com/pageContext#lifecycle.

I'm actually considering showing a warning when users do this.

Maybe there is a way to factor out the common code whitout needing to simulate/fake a pageContext?